File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/interfaceInJava/fieldsInInterface/Vehicle.java
package de.fraunhofer.iem.classesAndObjects.interfaceInJava.fieldsInInterface;

/**
 * Demonstrates the fields in Interface. In Java, fields in interface is always final
 *
 * @author Ranjith Krishnamurthy
 */
public interface Vehicle {
    String modelNumber = "Its Interface model number";
    String vehicleRegistrationNumber = "Its Interface vehicle registration number";

    public void start();

    public void stop();
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/interfaceInKotlin/propertiesInInterface/Vehicle.kt
package de.fraunhofer.iem.classesAndObjects.interfaceInKotlin.propertiesInInterface

/**
 * Demonstrates the properties in Interface
 *
 * @author Ranjith Krishnamurthy
 */
interface Vehicle {
    val modelNumber: String
    var vehicleRegistrationNumber: String

    fun start()
    fun stop()
}